### DISCUSSION TITLE
Toggle show/hide subtrees on Quiack Navigation Panel (QNP)

### DISCUSSION THREAD

**Discussion by zeroarst** [Able-Bear-Software/tab-nodes-tree#76] [Ideas / Feature Request]
There might be users like me who always have hundreds of tabs open to streamline their work without needing to bookmark them and reopen later. Providing a function to hide certain subtrees to focus on specific tasks could be useful.

For example, I have a `Work` group/branch tab node and a `Shopping` group/branch tab node. While working, I can temporarily hide the `Shopping` group/branch subtree to concentrate on my work.

Proposal:
1. Add an eye icon on the QNP to toggle the visibility of subtrees.
2. Display a "Some subtrees are hidden" message with a clear icon button on the right at the top if any subtrees are hidden.
3. Allow users to click the clear icon button to quickly show all subtrees.

Further thoughts are welcome!

**Comment by stefan123t** [comment]
While I agree that having some way of organising tabs into groups or work/subject would be awesome.
I do not know if having a separate visibility option for subtrees would really help that cause.
The reason being that I can already collapse / expand subtrees at will (e.g. using #62 even easily by double clicking the subtree header).

For me it would be interesting to have a pull down menu of a list of configurable topics / subjects fixed above the tab nodes tree.
I can then select a complete subtree and assign it to any of the configured topics or create a new topic to list it at.
It will be moved to the other topic and I can switch / decide to show and focus on a specific subject.
There should be a General / Default topic though. You could even define separate colors for your topics to distinguish them easily.

Maybe we need a shortcut to distignuish to open a new tab always on the default / the current or a specific topic. 
Something like Ctrl + Tab open of default group, Ctrl + Shift + Tab open in a specific / the current topic group.
